### PR TITLE
Add snippets to R2C

### PR DESCRIPTION
**Pull request by Rick-Anderson**
[Internal review URL](https://review.docs.microsoft.com/en-us/aspnet/core/web-api/route-to-code?view=aspnetcore-5.0&branch=pr-en-us-20276)

### CODE REVIEW

File: aspnetcore/web-api/route-to-code/sample/Startup2.cs [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+            {
+                endpoints.MapPost("/weather", async context =>
+                {
+                    if (!context.Request.HasJsonContentType())
+                    {
+                        context.Response.StatusCode = StatusCodes.Status415UnsupportedMediaType;
+                        return;
+                    }
+
+                  //var weather = await context.Request.ReadAsJsonAsync<WeatherForecast>();
```
Comment (Rick-Anderson) [inline aspnetcore/web-api/route-to-code/sample/Startup2.cs:48]: @JamesNK  your code, my code follow

File: aspnetcore/web-api/route-to-code/sample/Startup.cs [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+            // Endpoint aware middleware. 
+            // Middleware can use metadata from the matched endpoint.
+            app.UseAuthentication();
+            app.UseAuthorization();
+
+            // Execute the matched endpoint.
+            app.UseEndpoints(endpoints =>
+            {
+                // Configure this endpoint to not require a user.
```
Comment (JamesNK) [inline aspnetcore/web-api/route-to-code/sample/Startup.cs:40]:
```suggestion
                // This endpoint doesn't require authorization.
```

File: aspnetcore/web-api/route-to-code/sample/Startup2.cs (hunk 1 above)
Comment (JamesNK) [reply aspnetcore/web-api/route-to-code/sample/Startup2.cs:48]: Good catch
```suggestion
```

File: aspnetcore/web-api/route-to-code/sample/Startup2.cs [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+                    var weather = await context.Request.ReadFromJsonAsync<WeatherForecast>();
+                    await UpdateDatabase(weather);
```
Comment (JamesNK) [inline aspnetcore/web-api/route-to-code/sample/Startup2.cs:50]:
```suggestion
                    await UpdateDatabaseAsync(weather);
```

File: aspnetcore/web-api/route-to-code/sample/Startup2.cs [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+                    context.Response.StatusCode = StatusCodes.Status202Accepted;
+                });
+            });
+            #endregion
+
+        }
+
+        private Task UpdateDatabase(object weather)
```
Comment (JamesNK) [inline aspnetcore/web-api/route-to-code/sample/Startup2.cs:59]:
```suggestion
        private Task UpdateDatabaseAsync(object weather)
```

File: aspnetcore/web-api/route-to-code/sample/Startup3.cs [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+                });
+            });
+            #endregion
+
+        }
+
+        private Task UpdateDatabase(object weather)
+        {
+            throw new NotImplementedException();
+        }
```
Comment (JamesNK) [inline aspnetcore/web-api/route-to-code/sample/Startup3.cs:53]:
```suggestion
```

**Review by JamesNK — APPROVED** [review]

**Pull request merged**